src/core/public/index.ts

```diff
@@ -128,8 +128,8 @@ export type {
   NavigateToAppOptions,
   NavigateToUrlOptions,
 } from './application';
-
-export { SimpleSavedObjectImpl } from './saved_objects';
+// Export implementation only until we have a concrete mock for tests. See https://github.com/elastic/kibana/pull/137448 for more info
+export { SimpleSavedObjectImpl } from '@kbn/core-saved-objects-browser-internal';
```

Comment:
`SimpleSavedObjectImpl` isn't intended for use outside of the core domain. However,  three plugins depend on the concrete internal implementation for tests:

https://github.com/elastic/kibana/blob/755fad11be4bd920ad89da8e9533ad71a32a8253/src/plugins/saved_objects/public/saved_object/helpers/find_object_by_title.test.ts#L25-L32

https://github.com/elastic/kibana/blob/755fad11be4bd920ad89da8e9533ad71a32a8253/src/plugins/visualizations/public/utils/saved_objects_utils/find_object_by_title.test.ts#L25-L32

https://github.com/elastic/kibana/blob/755fad11be4bd920ad89da8e9533ad71a32a8253/x-pack/plugins/lens/public/persistence/saved_objects_utils/find_object_by_title.test.ts#L24-L26

We need a public mock for these so that we can truly keep the implementation internal to core, which can be handled as a follow-up task.